src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,243 @@
+                    CurrentSweepCycleStartTime = DateTime.MinValue,
+                    ContinuationToken = null
+                };
+                _scanInfo.Add(container, containerScanInfo);
+            }
+
+            containerScanInfo.Registrations.Add(triggerExecutor);
+        }
+
+        public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+
+            Task LogPollingTask = _pollLogStrategy.ExecuteAsync(cancellationToken);
+            List<IStorageBlob> failedNotifications = new List<IStorageBlob>();
+            List<Task> notifications = new List<Task>();
+
+            // Drain the background queue of blob written notifications.
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                IStorageBlob blob;
+
+                if (!_blobsFoundFromScanOrNotification.TryDequeue(out blob))
+                {
+                    break;
+                }
+
+                notifications.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
+            }
+            await Task.WhenAll(notifications);
+
+            List<Task> pollingTasks = new List<Task>();
+            pollingTasks.Add(LogPollingTask);
+
+            foreach (KeyValuePair<IStorageBlobContainer, ContainerScanInfo> containerScanInfoPair in _scanInfo)
+            {
+                pollingTasks.Add(PollAndNotify(containerScanInfoPair.Key, containerScanInfoPair.Value, cancellationToken, failedNotifications));
+            }
+
+            // Re-add any failed notifications for the next iteration.
+            foreach (IStorageBlob failedNotification in failedNotifications)
+            {
+                _blobsFoundFromScanOrNotification.Enqueue(failedNotification);
+            }
+            
+            await Task.WhenAll(pollingTasks);
+
+            // Run subsequent iterations at "_pollingInterval" second intervals.
+            return new TaskSeriesCommandResult(wait: Task.Delay(_pollintInterval));
+        }
+
+        private async Task PollAndNotify(IStorageBlobContainer container, ContainerScanInfo containerInfo, CancellationToken cancellationToken, List<IStorageBlob> failedNotifications)
+        {
+
+            cancellationToken.ThrowIfCancellationRequested();
+            IEnumerable<IStorageBlob> newBlobs = await PollNewBlobsAsync(container, containerInfo, cancellationToken);
+
+            foreach (IStorageBlob newBlob in newBlobs)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await NotifyRegistrationsAsync(newBlob, failedNotifications, cancellationToken);
+            }
+        }
+
+        public void Notify(IStorageBlob blobWritten)
+        {
+            ThrowIfDisposed();
+            _blobsFoundFromScanOrNotification.Enqueue(blobWritten);
+            _pollLogStrategy.Notify(blobWritten);
```

Comment:
Why do you also have to notify _pollLogsStrategy? Isn't adding the blob to the queue sufficient?